buildSrc/src/main/kotlin/com/saveourtool/osv4k/buildutils/PublishingConfiguration.kt

```diff
@@ -0,0 +1,162 @@
+        }
+    }
+
+    apply<MavenPublishPlugin>()
+    apply<SigningPlugin>()
+
+    configurePublications()
+
+    if (hasProperty("signingKey")) {
+        configureSigning()
+    }
+
+    // https://kotlinlang.org/docs/mpp-publish-lib.html#avoid-duplicate-publications
+    afterEvaluate {
+        val publicationsFromMainHost = listOf(
+            "jvm",
+            "js",
+            "linuxX64", "mingwX64", "macosX64",
+            "kotlinMultiplatform", "metadata",
+        )
+        configure<PublishingExtension> {
+            publications.matching { it.name in publicationsFromMainHost }.all {
+                val targetPublication = this@all
+                tasks.withType<AbstractPublishToMaven>()
+                    .matching { it.publication == targetPublication }
+                    .configureEach {
+                        onlyIf {
+                            // main publishing CI job is executed on Linux host
+                            DefaultNativePlatform.getCurrentOperatingSystem().isLinux.apply {
+                                if (!this) {
+                                    logger.lifecycle("Publication ${(it as AbstractPublishToMaven).publication.name} is skipped on current host")
+                                }
+                            }
+                        }
+                    }
+            }
+        }
+    }
+}
+
+@Suppress("TOO_LONG_FUNCTION", "GENERIC_VARIABLE_WRONG_DECLARATION")
+private fun Project.configurePublications() {
+    val dokkaJarProvider = tasks.register<Jar>("dokkaJar") {
+        group = "documentation"
+        archiveClassifier.set("javadoc")
+        from(tasks.findByName("dokkaHtml"))
+    }
+    configure<PublishingExtension> {
+        repositories {
+            mavenLocal()
+        }
+        publications.withType<MavenPublication>().forEach { publication ->
+            publication.artifact(dokkaJarProvider)
+            publication.pom {
+                name.set(project.name)
+                description.set(project.description ?: project.name)
+                url.set("https://github.com/saveourtool/save")
+                licenses {
+                    license {
+                        name.set("MIT License")
+                        url.set("http://www.opensource.org/licenses/mit-license.php")
+                        distribution.set("repo")
+                    }
+                }
+                developers {
+                    developer {
+                        id.set("petertrr")
+                        name.set("Petr Trifanov")
+                        email.set("peter.trifanov@gmail.com")
+                    }
+                    developer {
+                        id.set("akuleshov7")
+                        name.set("Andrey Kuleshov")
+                        email.set("andrewkuleshov7@gmail.com")
+                    }
+                }
+                scm {
+                    url.set("https://github.com/saveourtool/save")
```

Comment:
same

Reply:
may be this will be more actual and proper https://github.com/saveourtool/kompiledb/pull/35
https://github.com/saveourtool/kompiledb/pull/37/

Reply:
since Andrey investigated some problems in our current configureation, may be we did some improvements, please take a look

Reply:
thanks, will look

Reply:
took